File: Main2_51.java
import java.util.Scanner;

public class Main2_51 {

    static boolean canMake(int[] A, int index, int target) {
        if (target == 0)
            return true;
        if (index == A.length)
            return false;

        if (canMake(A, index + 1, target - A[index])) {
            return true;
        }

        return canMake(A, index + 1, target);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int[] A = new int[n];
        for (int i = 0; i < n; i++) {
            A[i] = sc.nextInt();
        }

        int q = sc.nextInt();
        for (int i = 0; i < q; i++) {
            int m = sc.nextInt();
            if (canMake(A, 0, m)) {
                System.out.println("yes");
            } else {
                System.out.println("no");
            }
        }

        sc.close();
    }
}

File: Main2_51.py
import sys
sys.setrecursionlimit(10**7)

def can_make(A, index, target):
    if target == 0:
        return True
    if index == len(A):
        return False

    if can_make(A, index + 1, target - A[index]):
        return True

    return can_make(A, index + 1, target)


def main():
    input = sys.stdin.readline

    n = int(input())
    A = list(map(int, input().split()))

    q = int(input())
    M = list(map(int, input().split()))

    for m in M:
        if can_make(A, 0, m):
            print("yes")
        else:
            print("no")


if __name__ == "__main__":
    main()